lib/podlet.js

```diff
@@ -818,6 +823,55 @@ export default class PodiumPodlet {
         return this.#view(incoming, data, ...args);
     }
 
+    /**
+     * I'm imaginging this as a helper method to be used in conjunction with render/podiumSend.
+     *
+     * Eg.
+     * app.get("/", (req, res) => {
+     *	res.podiumSend(res.podiumIsolate("<div>content to render</div>"));
+     * });
+     *
+     * This is for 2 reasons.
+     * 1. This gives the podlet developer full control of when to isolate the content without automagic.
+     * 2. This enables podiumSend/render to continue working exactly as they are making this a fully backwards compatible feature.
+     *
+     * We could consider integrating this feature into the render/podiumSend methods but this would require a rework of their method signatures to accommodate.
+     *
+     * We should consider naming of isolate. There may be better, more descriptive names? eg. podlet.wrapContent, podlet.wrapWithShadowDOM etc.
+     */
+    isolate(incoming, data, { mode = 'open' }) {
+        // This is an idea I had to allow the layout to disable podlet isolation via a context value.
+        if (incoming.context.isolate === false) {
+            return data;
+        }
+
+        // Ideally, when we create the DSD HTML tag for isolation, that isolation would be the podlet name to avoid the need to append or prepend naming elements.
+        // In order for that to be possible, the podlet name would need to follow custom element naming rules.
+        // An alternative strategy could be to attempt to cooerce the podlet name into a valid custom element name.
+        // Tricky for names that don't include a dash, underscore or camelcasing. Eg. foobar. How would we coorce that into a valid custom element name?
+        assert.ok(
+            customElementRegex.test(this.name),
+            'When using isolation, podlet.name must conform to custom element naming conventions: https://developer.mozilla.org/en-US/docs/Web/API/CustomElementRegistry/define#valid_custom_element_names',
+        );
+
+        // We can leverage the new asset scope feature and add a new scope type of 'shadow-dom' to make it possible to filter out shadow dom assets for inclusion.
```

Comment:
Is this a platform feature or something of ours? 🤔

Reply:
scope is a feature we added fairly recently. See https://github.com/podium-lib/podlet/pull/355